elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/EsParams.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.elasticsearch
+
+final class EsParams private (val indexName: String, val typeName: Option[String], val apiVersion: ApiVersion) {
+  def withIndexName(value: String): EsParams = {
+    require(value != null, "You must define an index name")
+
+    copy(indexName = value)
+  }
+
+  def withTypeName(value: String): EsParams = {
+    if (apiVersion == ApiVersion.V5) {
+      require(value != null && value.trim.nonEmpty, "You must define a type name for ElasticSearch API version V5")
+    }
```

Comment:
To make this check useful an API version should always be selected prior to calling the other setters.
I'd remove the default variants (an index name must always be given) and just keep the `V5` and `V7` constructors which can use `new` directly.
That's more invasive in our tests, but normal users will not switch between different versions in their code.

Or do you see a way that we don't need to have the API version on both `EsParams` and the source settings?

Reply:
I've changed `EsParams` to only have the 2 creation methods `V5` and `V7`. The only place where `EsParams.apiVersion` was used was in the `ElasticsearchSourceStage` and since that also has access to the `ElasticsearchSourceSettings` I've decided to remove `EsParams.apiVersion` all together.

Since both Java and Scala tests run for both V5 and V7 I did have to add a helper function in the tests to construct the right `EsParams` in the tests. The biggest issue with this setup is that all the documentation examples (`run-typed`, `run-jsobject`, `run-flow`, `string`, `kafka-example`, `custom-index-name-example` and `custom-search-params`  are taken from tests that now use the helper method to enable having a single test that runs against both API versions.

I'd prefer to have all examples documented with only 1 of the 2 `EsParam` versions (V7 since that'll be the default for Alpakka 3?). Do you see a better alternative to setting this up besides duplicating all these tests so that they work with the explicit `EsParams.V5` and `EsParams.V7`?

I did already add some generic documentation on `EsParams`.